### PR TITLE
Add mod_stream_management_backend

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "mnesia backend is hardcoded currently".

Proposed changes include:
* Pass HostType as an argument
* add backend logic
* move modules into stream_management directory

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108443) / small_tests / d7407ab6974e83bcbbdca0666d14966e602c74b1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3556/108443/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3556/108443/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-22_12.49.20/lib.mongooseim.logs/run.2022-02-22_12.49.21/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108601) / small_tests / 664a3458af5a2aed0371fec412d7c814432a5f14
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3556/108601/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3556/108601/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-22_15.27.57/lib.mongooseim.logs/run.2022-02-22_15.27.57/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108690) / small_tests / d8cc59d092199a18ae6b061a2daf01c89e4a4051
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3556/108690/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3556/108690/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-22_15.44.01/lib.mongooseim.logs/run.2022-02-22_15.44.01/suite.log.html)

File: src/stream_management/mod_stream_management.erl [hunk 1]
```diff
@@ -148,15 +142,12 @@ session_cleanup(Acc, _LUser, _LServer, _LResource, SID) ->
...
-        [] -> {error, smid_not_found};
-        [#sm_session{smid = SMID}] ->
-            mnesia:dirty_delete(sm_session, SMID),
-            case H of
-                undefined -> ok;
-                _ -> register_stale_smid_h(HostType, SMID, H)
-            end,
-            {ok, SMID}
+    MaybeSMID = unregister_smid(HostType, SID),
+    case MaybeSMID of
```
Comment (NelsonVides) [inline src/stream_management/mod_stream_management.erl:146]:
```suggestion
    case unregister_smid(HostType, SID) of
```

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

File: src/stream_management/mod_stream_management.erl (hunk 1 above)
Comment (arcusfelis) [reply src/stream_management/mod_stream_management.erl:146]: MaybeSMID is used below the case block though.

Comment (NelsonVides) [reply src/stream_management/mod_stream_management.erl:146]: Ah, didn't notice that xD
Ok so this PR seems perfect for me, can merge 👌🏽

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/108836) / small_tests / 6c21188ea5ca95bb72edff38c0e1b054adf02d38
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3556/108836/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3556/108836/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-23_09.57.55/lib.mongooseim.logs/run.2022-02-23_09.57.55/suite.log.html)

**Review by NelsonVides — COMMENTED** [review]
Seems good to me, but CI had two weird failures here 🤔

**Pull request merged**